url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/visualisation/charts/Generic_AgeGenderBoxPlot.java

```diff
@@ -0,0 +1,695 @@
+                    stats[4].subtract(stats[3]),
+                    getCellWidth(),
+                    0,
+                    getRoundingMode()).intValueExact();
+            int boxTopRow = coordinateToScreenRow(BigDecimal.valueOf(age + 1)) + 2;
+            //int boxTopRow = coordinateToScreenRow(BigDecimal.valueOf(age));
+            int boxMiddleRow = boxTopRow + (boxHeight / 2);
+            int boxBottomRow = boxTopRow + boxHeight;
+            int q1Col = coordinateToScreenCol(stats[3].negate());
+            int q3Col = coordinateToScreenCol(stats[4].negate());
+
+            // Draw min line
+            setPaint(Color.DARK_GRAY);
+            int minCol = coordinateToScreenCol(stats[6].negate());
+            abLine2D = new Line2D.Double(minCol, boxMiddleRow, q1Col, boxMiddleRow);
+            draw(abLine2D);
+            abLine2D = new Line2D.Double(
+                    minCol, boxMiddleRow + (whiskerHeight / 2),
+                    minCol, boxMiddleRow - (whiskerHeight / 2));
+            draw(abLine2D);
+
+            // Draw max line
+            setPaint(Color.DARK_GRAY);
+            int maxCol = coordinateToScreenCol(stats[7].negate());
+            abLine2D = new Line2D.Double(maxCol, boxMiddleRow, q3Col, boxMiddleRow);
+            draw(abLine2D);
+            abLine2D = new Line2D.Double(
+                    maxCol, boxMiddleRow + (whiskerHeight / 2),
+                    maxCol, boxMiddleRow - (whiskerHeight / 2));
+            draw(abLine2D);
+
+            // Draw box after drawing min and max lines to neaten overlaps
+            setPaint(Color.DARK_GRAY);
+            Rectangle2D r2 = new Rectangle2D.Double(
+                    q3Col,
+                    boxTopRow,
+                    boxWidth,
+                    boxHeight);
+            setPaint(Color.WHITE);
+            fillRect(
+                    q3Col,
+                    boxTopRow,
+                    boxWidth,
+                    boxHeight);
+            setPaint(Color.DARK_GRAY);
+            draw(r2);
+
+            // Draw median line
+            int medianCol = coordinateToScreenCol(stats[2].negate());
+            abLine2D = new Line2D.Double(medianCol, boxTopRow, medianCol, boxBottomRow);
+            draw(abLine2D);
+        }
+    }
+
+    @Override
+    public Object[] getDefaultData() {
+        int ageInterval = 5;
+        int startAgeOfEndYearInterval = 70;//95;
+        int decimalPlacePrecisionForCalculations = 10;
+        RoundingMode roundingMode = RoundingMode.HALF_UP;
+        Object[] data = getDefaultData(
+                ageInterval,
+                startAgeOfEndYearInterval,
+                decimalPlacePrecisionForCalculations,
+                roundingMode);
+        return data;
+    }
+
+    /**
+     * Override this method to use other data
+     *
+     * @param femalePopAge0
+     * @param malePopAge0
+     * @return
+     */
+    private static Object[] getPopulationData(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 351-393, lines 394-436 into separate methods.